packages/x-date-pickers/src/MultiSectionDigitalClock/MultiSectionDigitalClockSection.tsx

```diff
@@ -187,7 +187,9 @@ export const MultiSectionDigitalClockSection = React.forwardRef(
         {...other}
       >
         {items.map((option, index) => {
-          if (skipDisabled && option.isDisabled?.(option.value)) {
+          const isItemDisabled = option.isDisabled?.(option.value);
+          const isDisabled = disabled || isItemDisabled;
+          if (skipDisabled && isItemDisabled) {
```

Comment:
shouldn't this be `isDisabled`? 🤔 

```suggestion
          if (skipDisabled && isDisabled) {
```

Reply:
Not that such combination even makes sense—we would end up with empty list of items 😆 
That is already the case anyways, because `option.isDisabled` already does this check anyways.
But making sure and having the redundancy can't hurt. 👌

Reply:
Updated to check for `disabled` only here, removed the check on the `option` level. 👌